diff -up vlc-3.0.9.2/share/Makefile.am.2~ vlc-3.0.9.2/share/Makefile.am
--- vlc-3.0.9.2/share/Makefile.am.2~	2020-04-14 18:57:02.937481087 +0200
+++ vlc-3.0.9.2/share/Makefile.am	2020-04-14 18:57:25.732679027 +0200
@@ -97,7 +97,7 @@ skins2/default.vlt: $(skins2_default_vlt
 	GZIP=--no-name \
 	tar cvvzf skins2/default.vlt.tmp \
 		--format=ustar $$tarsort \
-		--owner=root --group=root --directory="$(srcdir)/skins2" \
+		--directory="$(srcdir)/skins2" \
 		default/
 	$(AM_V_at)mv -f -- skins2/default.vlt.tmp skins2/default.vlt
 
