--- vlc-3.0.20/modules/codec/avcodec/audio.c.orig	2024-04-11 22:04:48.840481336 +0100
+++ vlc-3.0.20/modules/codec/avcodec/audio.c	2024-04-11 22:56:54.658799280 +0100
@@ -65,7 +65,7 @@
     bool    b_extract;
     int     pi_extraction[AOUT_CHAN_MAX];
     int     i_previous_channels;
-    uint64_t i_previous_layout;
+    AVChannelLayout i_previous_layout;
 };
 
 #define BLOCK_FLAG_PRIVATE_REALLOCATED (1 << BLOCK_FLAG_PRIVATE_SHIFT)
@@ -139,7 +139,7 @@
     }
 
     ctx->sample_rate = p_dec->fmt_in.audio.i_rate;
-    ctx->channels = p_dec->fmt_in.audio.i_channels;
+    ctx->ch_layout.nb_channels = p_dec->fmt_in.audio.i_channels;
     ctx->block_align = p_dec->fmt_in.audio.i_blockalign;
     ctx->bit_rate = p_dec->fmt_in.i_bitrate;
     ctx->bits_per_coded_sample = p_dec->fmt_in.audio.i_bitspersample;
@@ -245,7 +245,7 @@
     p_sys->i_reject_count = 0;
     p_sys->b_extract = false;
     p_sys->i_previous_channels = 0;
-    p_sys->i_previous_layout = 0;
+    p_sys->i_previous_layout = (AVChannelLayout){0};
 
     /* */
     /* Try to set as much information as possible but do not trust it */
@@ -396,11 +396,11 @@
         if( ret == 0 )
         {
             /* checks and init from first decoded frame */
-            if( ctx->channels <= 0 || ctx->channels > INPUT_CHAN_MAX
+            if( ctx->ch_layout.nb_channels <= 0 || ctx->ch_layout.nb_channels > INPUT_CHAN_MAX
              || ctx->sample_rate <= 0 )
             {
                 msg_Warn( p_dec, "invalid audio properties channels count %d, sample rate %d",
-                          ctx->channels, ctx->sample_rate );
+                          ctx->ch_layout.nb_channels, ctx->sample_rate );
                 goto drop;
             }
             else if( p_dec->fmt_out.audio.i_rate != (unsigned int)ctx->sample_rate )
@@ -484,15 +484,15 @@
     /* Interleave audio if required */
     if( av_sample_fmt_is_planar( ctx->sample_fmt ) )
     {
-        p_block = block_Alloc(frame->linesize[0] * ctx->channels);
+        p_block = block_Alloc(frame->linesize[0] * ctx->ch_layout.nb_channels);
         if ( likely(p_block) )
         {
-            const void *planes[ctx->channels];
-            for (int i = 0; i < ctx->channels; i++)
+            const void *planes[ctx->ch_layout.nb_channels];
+            for (int i = 0; i < ctx->ch_layout.nb_channels; i++)
                 planes[i] = frame->extended_data[i];
 
             aout_Interleave(p_block->p_buffer, planes, frame->nb_samples,
-                            ctx->channels, p_dec->fmt_out.audio.i_format);
+                            ctx->ch_layout.nb_channels, p_dec->fmt_out.audio.i_format);
             p_block->i_nb_samples = frame->nb_samples;
         }
         av_frame_free(&frame);
@@ -511,7 +511,7 @@
         {
             aout_ChannelExtract( p_buffer->p_buffer,
                                  p_dec->fmt_out.audio.i_channels,
-                                 p_block->p_buffer, ctx->channels,
+                                 p_block->p_buffer, ctx->ch_layout.nb_channels,
                                  p_block->i_nb_samples, p_sys->pi_extraction,
                                  p_dec->fmt_out.audio.i_bitspersample );
             p_buffer->i_nb_samples = p_block->i_nb_samples;
@@ -580,33 +580,36 @@
     p_dec->fmt_out.audio.i_rate = p_sys->p_context->sample_rate;
 
     /* */
-    if( p_sys->i_previous_channels == p_sys->p_context->channels &&
-        p_sys->i_previous_layout == p_sys->p_context->channel_layout )
+    if( p_sys->i_previous_channels == p_sys->p_context->ch_layout.nb_channels &&
+        av_channel_layout_compare(&p_sys->i_previous_layout,&p_sys->p_context->ch_layout) == 0 )
         return;
     if( b_trust )
     {
-        p_sys->i_previous_channels = p_sys->p_context->channels;
-        p_sys->i_previous_layout = p_sys->p_context->channel_layout;
+        p_sys->i_previous_channels = p_sys->p_context->ch_layout.nb_channels;
+        av_channel_layout_copy(&p_sys->i_previous_layout, &p_sys->p_context->ch_layout);
     }
 
     const unsigned i_order_max = sizeof(pi_channels_map)/sizeof(*pi_channels_map);
     uint32_t pi_order_src[i_order_max];
 
     int i_channels_src = 0;
-    int64_t channel_layout =
-        p_sys->p_context->channel_layout ? p_sys->p_context->channel_layout :
-        av_get_default_channel_layout( p_sys->p_context->channels );
+    AVChannelLayout channel_layout;
+    if(p_sys->p_context->ch_layout.u.mask) {
+       av_channel_layout_copy(&channel_layout, &p_sys->p_context->ch_layout);
+    } else {
+       av_channel_layout_default(&channel_layout, p_sys->p_context->ch_layout.nb_channels );
+    }
 
-    if( channel_layout )
+    if(av_channel_layout_check(&channel_layout))
     {
         for( unsigned i = 0; i < i_order_max
-         && i_channels_src < p_sys->p_context->channels; i++ )
+         && i_channels_src < p_sys->p_context->ch_layout.nb_channels; i++ )
         {
-            if( channel_layout & pi_channels_map[i][0] )
+            if( channel_layout.u.mask & pi_channels_map[i][0] )
                 pi_order_src[i_channels_src++] = pi_channels_map[i][1];
         }
 
-        if( i_channels_src != p_sys->p_context->channels && b_trust )
+        if( i_channels_src != p_sys->p_context->ch_layout.nb_channels && b_trust )
             msg_Err( p_dec, "Channel layout not understood" );
 
         /* Detect special dual mono case */
@@ -638,7 +641,7 @@
     {
         msg_Warn( p_dec, "no channel layout found");
         p_dec->fmt_out.audio.i_physical_channels = 0;
-        p_dec->fmt_out.audio.i_channels = p_sys->p_context->channels;
+        p_dec->fmt_out.audio.i_channels = p_sys->p_context->ch_layout.nb_channels;
     }
 
     aout_FormatPrepare( &p_dec->fmt_out.audio );
--- vlc-3.0.20/modules/codec/avcodec/fourcc.c.orig	2024-04-11 23:18:24.472481862 +0100
+++ vlc-3.0.20/modules/codec/avcodec/fourcc.c	2024-04-11 23:17:13.105163354 +0100
@@ -182,7 +182,7 @@
     /* AV_CODEC_ID_V210X */
     { VLC_CODEC_TMV, AV_CODEC_ID_TMV },
     { VLC_CODEC_V210, AV_CODEC_ID_V210 },
-#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT( 54, 50, 100 ) && LIBAVCODEC_VERSION_MICRO >= 100
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT( 54, 50, 100 ) && LIBAVCODEC_VERSION_MICRO >= 100 && LIBAVCODEC_VERSION_MAJOR < 61
     { VLC_CODEC_VUYA, AV_CODEC_ID_AYUV },
 #endif
     /* AV_CODEC_ID_DPX */
--- vlc-3.0.20/modules/codec/avcodec/encoder.c.orig	2024-04-11 23:16:32.476696552 +0100
+++ vlc-3.0.20/modules/codec/avcodec/encoder.c	2024-04-11 23:38:17.189055804 +0100
@@ -739,9 +739,10 @@
         date_Set( &p_sys->buffer_date, AV_NOPTS_VALUE );
         p_context->time_base.num = 1;
         p_context->time_base.den = p_context->sample_rate;
-        p_context->channels      = p_enc->fmt_out.audio.i_channels;
+        p_context->ch_layout.order            = AV_CHANNEL_ORDER_NATIVE;
+        p_context->ch_layout.nb_channels      = p_enc->fmt_out.audio.i_channels;
 #if LIBAVUTIL_VERSION_CHECK( 52, 2, 6, 0, 0)
-        p_context->channel_layout = channel_mask[p_context->channels][1];
+        p_context->ch_layout.u.mask = channel_mask[p_context->ch_layout.nb_channels][1];
 
         /* Setup Channel ordering for multichannel audio
          * as VLC channel order isn't same as libavcodec expects
@@ -752,17 +753,17 @@
         /* Specified order
          * Copied from audio.c
          */
-        const unsigned i_order_max = 8 * sizeof(p_context->channel_layout);
+        const unsigned i_order_max = 8 * sizeof(p_context->ch_layout.u.mask);
         uint32_t pi_order_dst[AOUT_CHAN_MAX] = { };
         uint32_t order_mask = 0;
         int i_channels_src = 0;
 
-        if( p_context->channel_layout )
+        if( p_context->ch_layout.u.mask )
         {
             msg_Dbg( p_enc, "Creating channel order for reordering");
             for( unsigned i = 0; i < sizeof(pi_channels_map)/sizeof(*pi_channels_map); i++ )
             {
-                if( p_context->channel_layout & pi_channels_map[i][0] )
+                if( p_context->ch_layout.u.mask & pi_channels_map[i][0] )
                 {
                     msg_Dbg( p_enc, "%d %"PRIx64" mapped to %"PRIx64"", i_channels_src, pi_channels_map[i][0], pi_channels_map[i][1]);
                     pi_order_dst[i_channels_src++] = pi_channels_map[i][1];
@@ -774,7 +775,7 @@
         {
             msg_Dbg( p_enc, "Creating default channel order for reordering");
             /* Create default order  */
-            for( unsigned int i = 0; i < __MIN( i_order_max, (unsigned)p_sys->p_context->channels ); i++ )
+            for( unsigned int i = 0; i < __MIN( i_order_max, (unsigned)p_sys->p_context->ch_layout.nb_channels ); i++ )
             {
                 if( i < sizeof(pi_channels_map)/sizeof(*pi_channels_map) )
                 {
@@ -916,10 +917,10 @@
             goto error;
         }
 
-        if( p_context->channels > 2 )
+        if( p_context->ch_layout.nb_channels > 2 )
         {
-            p_context->channels = 2;
-            p_context->channel_layout = channel_mask[p_context->channels][1];
+            p_context->ch_layout.nb_channels = 2;
+            p_context->ch_layout.u.mask = channel_mask[p_context->channels][1];
 
             /* Change fmt_in in order to ask for a channels conversion */
             p_enc->fmt_in.audio.i_channels =
@@ -1277,8 +1278,7 @@
     av_frame_unref( p_sys->frame );
     p_sys->frame->format     = p_sys->p_context->sample_fmt;
     p_sys->frame->nb_samples = leftover_samples + p_sys->i_samples_delay;
-    p_sys->frame->channel_layout = p_sys->p_context->channel_layout;
-    p_sys->frame->channels = p_sys->p_context->channels;
+    av_channel_layout_copy(&p_sys->frame->ch_layout, &p_sys->p_context->channel_layout);
 
     p_sys->frame->pts        = date_Get( &p_sys->buffer_date ) * p_sys->p_context->time_base.den /
                                 CLOCK_FREQ / p_sys->p_context->time_base.num;
@@ -1408,11 +1408,10 @@
         p_sys->frame->pts        = date_Get( &p_sys->buffer_date ) * p_sys->p_context->time_base.den /
                                     CLOCK_FREQ / p_sys->p_context->time_base.num;
 
-        p_sys->frame->channel_layout = p_sys->p_context->channel_layout;
-        p_sys->frame->channels = p_sys->p_context->channels;
+        av_channel_layout_copy(&p_sys->frame->ch_layout, &p_sys->p_context->channel_layout);
 
         const int in_bytes = p_sys->frame->nb_samples *
-            p_sys->p_context->channels * p_sys->i_sample_bytes;
+            p_sys->p_context->ch_layout.nb_channels * p_sys->i_sample_bytes;
 
         if( p_sys->b_planar )
         {
@@ -1451,7 +1450,7 @@
     if( p_aout_buf->i_nb_samples > 0 )
     {
        memcpy( p_sys->p_buffer + buffer_delay, p_aout_buf->p_buffer,
-               p_aout_buf->i_nb_samples * p_sys->i_sample_bytes * p_sys->p_context->channels);
+               p_aout_buf->i_nb_samples * p_sys->i_sample_bytes * p_sys->p_context->ch_layout.nb_channels);
        p_sys->i_samples_delay += p_aout_buf->i_nb_samples;
     }
 
