diff -up vlc-master/share/Makefile.am.2~ vlc-master/share/Makefile.am
--- vlc-master/share/Makefile.am.2~	2023-11-19 16:15:08.893550246 +0100
+++ vlc-master/share/Makefile.am	2023-11-19 16:16:05.589307763 +0100
@@ -154,7 +154,7 @@ skins2/default.vlt: $(skins2_default_vlt
 	$(AM_V_GEN)GZIP=--no-name \
 	tar cvvzf skins2/default.vlt.tmp \
 		--format=ustar $$tarsort \
-		--owner=root --group=root --directory="$(srcdir)/skins2" \
+		--directory="$(srcdir)/skins2" \
 		$$(cd "$(srcdir)/skins2" && find default/ -type f | LC_ALL=C sort)
 	$(AM_V_at)mv -f -- skins2/default.vlt.tmp skins2/default.vlt
 
