--- vlc-3.0.8/modules/codec/webvtt/CSSGrammar.c	2019-08-19 08:39:39.000000000 +0200
+++ vlc-3.0.8/modules/codec/webvtt/CSSGrammar.c-new	2019-08-25 01:20:26.598129957 +0200
@@ -310,7 +310,7 @@ typedef short yytype_int16;
 # define YYUSE(E) /* empty */
 #endif
 
-#if defined __GNUC__ && ! defined __ICC && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
+#if !defined(__clang__) && defined __GNUC__ && 407 <= __GNUC__ * 100 + __GNUC_MINOR__
 /* Suppress an incorrect diagnostic about yylval being uninitialized.  */
 # define YY_IGNORE_MAYBE_UNINITIALIZED_BEGIN \
     _Pragma ("GCC diagnostic push") \
