--- vlc-3.0.11.1/modules/access/live555.cpp.orig	2020-12-30 19:12:05.372361483 -0500
+++ vlc-3.0.11.1/modules/access/live555.cpp	2020-12-30 19:11:28.271611062 -0500
@@ -720,6 +720,7 @@ static int SessionsSetup( demux_t *p_dem
     demux_sys_t             *p_sys  = p_demux->p_sys;
     MediaSubsessionIterator *iter   = NULL;
     MediaSubsession         *sub    = NULL;
+    struct sockaddr_storage fSubAddr;
 
     bool           b_rtsp_tcp;
     int            i_client_port;
@@ -852,7 +853,8 @@ static int SessionsSetup( demux_t *p_dem
             if( !p_sys->b_multicast )
             {
                 /* We need different rollover behaviour for multicast */
-                p_sys->b_multicast = IsMulticastAddress( sub->connectionEndpointAddress() );
+                sub->getConnectionEndpointAddress( fSubAddr );
+                p_sys->b_multicast = IsMulticastAddress( fSubAddr );
             }
 
             tk = (live_track_t*)malloc( sizeof( live_track_t ) );
