diff -up vlc-3.0.3/modules/gui/qt/components/simple_preferences.cpp.omv~ vlc-3.0.3/modules/gui/qt/components/simple_preferences.cpp
--- vlc-3.0.3/modules/gui/qt/components/simple_preferences.cpp.omv~	2018-07-31 21:35:15.241458890 +0200
+++ vlc-3.0.3/modules/gui/qt/components/simple_preferences.cpp	2018-07-31 21:35:24.606512652 +0200
@@ -39,6 +39,7 @@
 #include <QSignalMapper>
 #include <QVBoxLayout>
 #include <QScrollArea>
+#include <QButtonGroup>
 
 #include <QStyleFactory>
 #include <QSettings>
