diff -Naur vlc-3.0.9.2 vlc-3.0.9.2-om       diff -Naur vlc-3.0.9.2/share/Makefile.am vlc-3.0.9.2-om/share/Makefile.am
--- vlc-3.0.9.2/share/Makefile.am       2020-04-05 15:46:34.000000000 +0200
+++ vlc-3.0.9.2-om/share/Makefile.am    2020-04-10 16:08:35.776402334 +0200
@@ -97,7 +97,7 @@
        GZIP=--no-name \
        tar cvvzf skins2/default.vlt.tmp \
                --format=ustar $$tarsort \
-               --owner=root --group=root --directory="$(srcdir)/skins2" \
+               --directory="$(srcdir)/skins2" \
                default/
        $(AM_V_at)mv -f -- skins2/default.vlt.tmp skins2/default.vlt
 
diff -Naur vlc-3.0.9.2/share/Makefile.in vlc-3.0.9.2-om/share/Makefile.in
--- vlc-3.0.9.2/share/Makefile.in       2020-04-09 12:38:22.000000000 +0200
+++ vlc-3.0.9.2-om/share/Makefile.in    2020-04-10 16:09:47.250490950 +0200
@@ -1845,7 +1845,7 @@
        GZIP=--no-name \
        tar cvvzf skins2/default.vlt.tmp \
                --format=ustar $$tarsort \
-               --owner=root --group=root --directory="$(srcdir)/skins2" \
+               --directory="$(srcdir)/skins2" \
                default/
        $(AM_V_at)mv -f -- skins2/default.vlt.tmp skins2/default.vlt
 
